File: responses.py
import discord
from discord.ext import commands

# Contador de xd (lectura del archivo)
def read_counter(server_id: int):
    counters = {}
    try:
        with open('xd_counter.txt', 'r') as file:
            for line in file:
                if ':' in line:
                    server_id_from_file, count = line.strip().split(':')
                    counters[int(server_id_from_file)] = int(count)
    except FileNotFoundError:
        pass
    
    return counters.get(server_id, 0)

def write_counter(server_id: int, count: int):
    counters = {}
    try:
        with open('xd_counter.txt', 'r') as file:
            for line in file:
                if ':' in line:
                    server_id_from_file, count_from_file = line.strip().split(':')
                    counters[int(server_id_from_file)] = int(count_from_file)
    except FileNotFoundError:
        pass

    counters[server_id] = count
    with open('xd_counter.txt', 'w') as file:
        for server_id, count in counters.items():
            file.write(f"{server_id}:{count}\n")

def setup_responses(bot: commands.Bot):
    @bot.event
    async def on_message(message: discord.Message):
        if message.author == bot.user:
            return

        if message.content and not message.attachments:
            content = message.content.strip().lower()

        # Responder a mensajes con una cadena de 'a'
        if content == 'a' * len(content):
            count_a = len(content)
            response = 'rr' + ('o' * count_a) + 'z .¿'
            await message.channel.send(response)
            print(f"Se envio '{response}' en el server '{message.guild.name}'")
            return

        # Responder a ciertas palabras clave
        if content.lower() == "que":
            response = "so. Te trolie jeje nya"
            await message.channel.send(response)
            print(f"Se envio '{response}' en el server '{message.guild.name}'")
            return

        if content.lower() == "real":
            response = "real real real"
            await message.channel.send(response)
            print(f"Se envio '{response}' en el server '{message.guild.name}'")
            return

        if 'xd' in content:
            server_id = message.guild.id
            current_count = read_counter(server_id)
            new_count = current_count + 1
            write_counter(server_id, new_count)
            await message.channel.send(f"El xD ha sido enviado {new_count} veces hasta ahora.")
            print(f"Se actualizo el XD a {new_count} en el server {message.guild.name}'")
            return

        # Si el usuario menciona al bot @Ressy
        if bot.user.mentioned_in(message):
            lower_content = message.content.lower()

            if any(word in lower_content for word in ["te amo", "te quiero", "cariño"]):
                await message.channel.send("Y-yo tambien te quiero mucho nwn.")
                return
            
            if any(word in lower_content for word in ["hola", "hi", "hey", "holi"]):
                await message.channel.send("¡Hola! ¿Cómo estás? (≧ω≦)/")
                return

            if any(word in lower_content for word in ["adiós", "bye", "chao", "chau", "ya me voy"]):
                await message.channel.send("¡Adiós! ¡Cuídate mucho! ( ˘︹˘ )")
                return

            if any(word in lower_content for word in ["gracias", "thanks", "thank you", "thx"]):
                await message.channel.send("¡De nada! Estoy aquí para ayudarte. (◕‿◕✿)")
                return

            if any(word in lower_content for word in ["como estás", "que tal", "como vas", "como andas"]):
                await message.channel.send("¡Estoy genial! ¿Y tú? (◠‿◠✿)")
                return

            if any(word in lower_content for word in ["ayuda", "help"]):
                await message.channel.send("¡Estoy aquí para ayudarte! ¿Qué necesitas? (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧")
                return

            if any(word in lower_content for word in ["broma", "joke"]):
                await message.channel.send("¿Sabías que los computadores nunca se cansan? ¡Porque tienen mucho 'megaherz'! (≧∇≦)")
                return

            if any(word in lower_content for word in ["feliz", "happy", "alegre"]):
                await message.channel.send("¡Me alegra que estés feliz! (＾▽＾)")
                return

            if any(word in lower_content for word in ["triste", "sad", "deprimido", "desanimado"]):
                await message.channel.send("Oh no, ¡ánimo! ¡Todo va a estar bien! (◕︿◕✿)")
                return
        else:
            return
        # Asegúrate de que los comandos se procesen también
        await bot.process_commands(message)

File: modules/responses.js
import fs from 'fs';

function readCounter(serverId) {
    const counters = {};
    try {
        const data = fs.readFileSync('xd_counter.txt', 'utf8');
        data.split('\n').forEach(line => {
            if (line.includes(':')) {
                const [serverIdFromFile, count] = line.trim().split(':');
                counters[serverIdFromFile] = parseInt(count, 10);
            }
        });
    } catch (error) {
        console.error("No se pudo leer el archivo xd_counter.txt:", error);
    }
    return counters[serverId] || 0;
}

function writeCounter(serverId, count) {
    const counters = {};
    try {
        const data = fs.readFileSync('xd_counter.txt', 'utf8');
        data.split('\n').forEach(line => {
            if (line.includes(':')) {
                const [serverIdFromFile, countFromFile] = line.trim().split(':');
                counters[serverIdFromFile] = parseInt(countFromFile, 10);
            }
        });
    } catch (error) {
        console.error("No se pudo leer el archivo xd_counter.txt:", error);
    }

    counters[serverId] = count;
    const newData = Object.entries(counters).map(([id, cnt]) => `${id}:${cnt}`).join('\n');
    fs.writeFileSync('xd_counter.txt', newData);
}

function setupResponses(client) {
    client.on('messageCreate', async message => {
        if (message.author.bot) return;

        const content = message.content.trim().toLowerCase();

        if (/^a+$/.test(content)) {
            const response = 'rr' + 'o'.repeat(content.length) + 'z .¿';
            await message.channel.send(response);
            console.log(`Se envió '${response}' en el servidor '${message.guild.name}'`);
            return;
        }

        if (content === "que") {
            await message.channel.send("so. Te trolie jeje nya");
            return;
        }

        if (content === "real") {
            await message.channel.send("real real real");
            return;
        }

        if (content.includes('xd')) {
            const serverId = message.guild.id;
            const currentCount = readCounter(serverId);
            const newCount = currentCount + 1;
            writeCounter(serverId, newCount);
            await message.channel.send(`El xD ha sido enviado ${newCount} veces hasta ahora.`);
            console.log(`Se actualizó el XD a ${newCount} en el servidor '${message.guild.name}'`);
            return;
        }

        if (message.mentions.has(client.user)) {
            const mentionReplies = {
                teQuiero: ["te amo", "te quiero", "cariño"],
                hola: ["hola", "hi", "hey", "holi"],
                adios: ["adiós", "bye", "chao", "chau", "ya me voy"],
                gracias: ["gracias", "thanks", "thank you", "thx"],
                comoEstas: ["como estas", "como estás","que tal", "como vas", "como andas"],
                ayuda: ["ayuda", "help"],
                broma: ["broma", "joke"],
                feliz: ["feliz", "happy", "alegre"],
                triste: ["triste", "sad", "deprimido", "desanimado"]
            };

            for (const [key, triggers] of Object.entries(mentionReplies)) {
                if (triggers.some(word => content.includes(word))) {
                    let reply;
                    switch (key) {
                        case 'teQuiero': reply = "Y-yo tambien te quiero mucho nwn."; break;
                        case 'hola': reply = "¡Hola! ¿Cómo estás? (≧ω≦)/"; break;
                        case 'adios': reply = "¡Adiós! ¡Cuídate mucho! ( ˘︹˘ )"; break;
                        case 'gracias': reply = "¡De nada! Estoy aquí para ayudarte. (◕‿◕✿)"; break;
                        case 'comoEstas': reply = "¡Estoy genial! ¿Y tú? (◠‿◠✿)"; break;
                        case 'ayuda': reply = "¡Estoy aquí para ayudarte! ¿Qué necesitas? (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧"; break;
                        case 'broma': reply = "¿Sabías que los computadores nunca se cansan? ¡Porque tienen mucho 'megaherz'! (≧∇≦)"; break;
                        case 'feliz': reply = "¡Me alegra que estés feliz! (＾▽＾)"; break;
                        case 'triste': reply = "Oh no, ¡ánimo! ¡Todo va a estar bien! (◕︿◕✿)"; break;
                    }
                    await message.channel.send(reply);
                    return;
                }
            }
        }
    });
}

export { setupResponses };
